### PR TITLE
i18n(de): update translation reference/overrides

### PR DESCRIPTION
**Pull request by trueberryless**
#### Description

Translation of `reference/overrides.mdx` into German language.

### CODE REVIEW

**Comment by trueberryless** [comment]
Link fixed in #2459

File: docs/src/content/docs/de/reference/overrides.md [hunk 1]
```diff
@@ -18,59 +20,73 @@ Um deine eigenen Komponenten zu schreiben, importiere den `Props`-Typ von Starli
...
 BCP-47-Sprachkennzeichen für das Gebietsschema dieser Seite, z.B. `de`, `zh-CN` oder `pt-BR`.
 
 #### `locale`
 
-**Type:** `string | undefined`
+**Typ:** `string | undefined`
 
 Der Basispfad, unter dem eine Sprache angeboten wird. `undefined` für Root-Locale-Slugs.
 
+#### `siteTitle`
```
Comment (HiDeoo) [inline docs/src/content/docs/de/reference/overrides.md:55]: I cannot suggest an edit for it as it's outside of the review range but this change is missing the anchor change from https://github.com/withastro/starlight/pull/1620/files#diff-ad80506c0d18072cbd9a85e2090c2be41e5a5523fe9f2991a1c653e41d089591, the `-1` addition to properly link to the component and not the prop.

File: docs/src/content/docs/de/reference/overrides.md [hunk 2]
```diff
@@ -1,8 +1,10 @@
 ---
-title: Komponenten-Ersetzung Referenz
-description: Ein Überblick über die Komponenten und Komponentenprops, die von Starlight Ersetzungen unterstützt werden.
+title: Ersetzung von Komponenten Referenz
```
Comment (HiDeoo) [inline docs/src/content/docs/de/reference/overrides.md:2]: Related discussion for reference https://github.com/withastro/starlight/pull/2470/files/2b486d1b1fa4acf3f0c87da750f3ce219794f78e#diff-9185fd84aba7e9a14b394be96e2bd3562baac10c0285be708a3109c5cbe72802

File: docs/src/content/docs/de/reference/icons.mdx [hunk 3]
```diff
@@ -1,6 +1,8 @@
 ---
 title: Symbole Referenz
 description: Eine Übersicht über alle in Starlight verfügbaren Symbole.
+sidebar:
+  label: Symbole
```
Comment (HiDeoo) [inline docs/src/content/docs/de/reference/icons.mdx:5]: Related disccussion for reference https://github.com/withastro/starlight/pull/2470/files/2b486d1b1fa4acf3f0c87da750f3ce219794f78e#diff-9185fd84aba7e9a14b394be96e2bd3562baac10c0285be708a3109c5cbe72802

**Review by HiDeoo — CHANGES_REQUESTED** [review]
Thanks for the new translation :thumbsup:

As a non-native speaker, I reviewed this PR using the [Docs Translation Review Checklist](https://contribute.docs.astro.build/reviewers/reviewing-translations/). I left one comment regarding a missing change and linked to the discussion regarding the sidebar label changes for reference until we know which style should be used.

File: docs/src/content/docs/de/reference/overrides.md (hunk 1 above)
Comment (trueberryless) [reply docs/src/content/docs/de/reference/overrides.md:55]: I got u 😊

**Comment by trueberryless** [comment]
Thanks for the review, should be good to go when links are ready!

**Review by HiDeoo — APPROVED** [review]
And links are now fixed, everything looks in order, thanks for the updates :rocket:

**Pull request merged**